extensions_userprovided/raspiBackupDiscordWrapper.sh

```
@@ -0,0 +1,241 @@
+# DISC_DESC
+# DISC_COLOR
+#
+# these variables are set in function send_discord
+###########################################
+function build_json() {
+        DISCORD_JSON=$(jq \
+                        --arg cont "$DISC_CONTENT" \
+                        --arg title "$DISC_TITLE" \
+                        --arg color "$DISC_COLOR" \
+                        --arg desc "$DISC_DESC" \
+                       '.content=$cont | .embeds[0].title=$title | .embeds[0].description=$desc | .embeds[0].color=$color' <<< "$DEFAULTJSON")
+}
+###########################################
+### Function to set variables for JSON message, and calls JSON construction function: build_json
+# it then send the message to discord using CURL
+# $1 contains the raspiBackup return code in order to adapt the message embed's color
+###########################################
+function send_discord() { # $1:status of the operation, 0=) success, not 0 is failure
+        local discord_color=$CONFIG_COLOR_DEFAULT
+        if [[ $1 -ne  0 ]]; then
+                discord_color=$CONFIG_COLOR_FAILURE
+                STATUS="FAILURE. Error code is: $1"
+                TITLE_SUFFIX="failed !!!"
+        else
+                discord_color=$CONFIG_COLOR_SUCCESS
+                STATUS="SUCCESS."
+                TITLE_SUFFIX="finished successfully."
+        fi
+        DISC_CONTENT="Raspi Backup Status: $(date)"
+        DISC_TITLE="Raspi Backup $TITLE_SUFFIX"
+	# just to ensure not to fail in case the message file is not found and send a minimal status message without details
+	[ -z "$MSG_CONTENT_FILE" ] && DISC_DESC="The raspiBackup operation just finished with the status: $STATUS" || DISC_DESC="The raspiBackup operation just finished with the status: $STATUS $(<"$MSG_CONTENT_FILE")"
+
+        DISC_COLOR=$discord_color
+
+        build_json
+
+        DISCORD_URL=$CONFIG_DISCORD_URL
+
+        curl -s -H "Content-Type: application/json" -X POST -d "$DISCORD_JSON" $DISCORD_URL > /dev/null
```

Is there any reason you suppress any error messages?

---

cursl is sending 'progress' and other stuff on console, which I wanted to avoid. If curl fails, this is not a big deal.